### PR TITLE
feat: Adds custom pub/sub plugin

### PR DESCRIPTION
**Pull request by danmrichards**
Adds a custom version of the pub/sub plugin that loads it's credentials from an encrypted string in config

### CODE REVIEW

File: plugins/outputs/mp_cloud_pubsub/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+as one of the supported [output data formats][].
+
+It is based upon the standard Telegraf [cloud_pubsub][] plugin but loads the
+credentials for Google Pub/Sub from an encrypted string in the config file.
+
+
+### Configuration
+
+This section contains the default TOML to configure the plugin.  You can
+generate it using `telegraf --usage cloud_pubsub`.
```
Comment (darkliquid) [inline plugins/outputs/mp_cloud_pubsub/README.md:13]: shouldn't this be?

```suggestion
generate it using `telegraf --usage mp_cloud_pubsub`.
```

File: plugins/outputs/mp_cloud_pubsub/pubsub_test.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
+package mp_cloud_pubsub
+
+import (
+	"testing"
+
+	"encoding/base64"
```
Comment (darkliquid) [inline plugins/outputs/mp_cloud_pubsub/pubsub_test.go:6]:
```suggestion
	"encoding/base64"
	"testing"
```

File: plugins/outputs/mp_cloud_pubsub/topic_stubbed.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+import (
+	"context"
+	"errors"
+	"fmt"
+	"runtime"
+	"sync"
+	"testing"
+	"time"
+
+	"encoding/base64"
```
Comment (darkliquid) [inline plugins/outputs/mp_cloud_pubsub/topic_stubbed.go:12]:
```suggestion
	"context"
	"encoding/base64"
	"errors"
	"fmt"
	"runtime"
	"sync"
	"testing"
	"time"
```

File: plugins/outputs/mp_cloud_pubsub/README.md (hunk 1 above)
Comment (danmrichards) [reply plugins/outputs/mp_cloud_pubsub/README.md:13]: Nice catch

File: plugins/outputs/mp_cloud_pubsub/pubsub_test.go (hunk 2 above)
Comment (danmrichards) [reply plugins/outputs/mp_cloud_pubsub/pubsub_test.go:6]: Nice spot, yeah this is actually wrong in the original Telegraf plugin too :smile:

File: plugins/outputs/mp_cloud_pubsub/topic_stubbed.go (hunk 3 above)
Comment (danmrichards) [reply plugins/outputs/mp_cloud_pubsub/topic_stubbed.go:12]: Nice spot, yeah this is actually wrong in the original Telegraf plugin too :smile:

**Pull request merged**